# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/AudioSinkWrapper.h
# Commit: 569b8d3a9fcd
# Full Hash: 569b8d3a9fcd96eb276c5225f81fcf9c01183565
# Author: Paul Adenot <paul@paul.cx>
# Date: 2022-05-10 03:50:31
# Regressor Bug: 1743834
# File Overlap Count: 1
# Description:
#   Bug 1743834 - Split initialization and start in AudioSinkWrapper. r=alwu
#   
#   No change in functionality, this is to prepare for the next patch that will do
#   the initialization asynchronously because it's expensing and blocking.
#   
# ==============================================================================

diff -r 9c37cd8faac3 -r 569b8d3a9fcd dom/media/mediasink/AudioSinkWrapper.h
--- a/dom/media/mediasink/AudioSinkWrapper.h	Mon May 09 13:20:52 2022 +0000
+++ b/dom/media/mediasink/AudioSinkWrapper.h	Mon May 09 13:20:53 2022 +0000
@@ -30,7 +30,7 @@
   class Creator {
    public:
     virtual ~Creator() = default;
-    virtual AudioSink* Create(const media::TimeUnit& aStartTime) = 0;
+    virtual AudioSink* Create() = 0;
   };
 
   // Wrap around a function object which creates AudioSinks.
@@ -38,9 +38,7 @@
   class CreatorImpl : public Creator {
    public:
     explicit CreatorImpl(const Function& aFunc) : mFunction(aFunc) {}
-    AudioSink* Create(const media::TimeUnit& aStartTime) override {
-      return mFunction(aStartTime);
-    }
+    AudioSink* Create() override { return mFunction(); }
 
    private:
     Function mFunction;
